example/lib/pages/post_detail/post_detail_page_scenes.dart

```diff
@@ -9,23 +9,20 @@ import 'post_detail_page.dart';
 class PostDetailPageScene extends Scene {
   static const String _currentPostKey = 'currentPost';
 
-  @override
-  Future<void> setUp(EnvironmentState environmentState) async {
-    super.setUp(environmentState);
-    environmentState.setDefault(_currentPostKey, Post.fakePosts().first);
-  }
-
   @override
   String get title => 'Post Detail';
 
   /// This [Scene] overrides the otional [environmentControlBuilders] getter to
   /// add a custom control to the Stager environment control panel.
   @override
-  List<EnvironmentControlBuilder> get environmentControlBuilders =>
-      <EnvironmentControlBuilder>[
-        (_, EnvironmentState state) {
-          return DropdownControl<Post>(
-              value: state.get<Post>(_currentPostKey)!,
+  List<EnvironmentControl<dynamic>> get environmentControls =>
+      <EnvironmentControl<dynamic>>[
+        EnvironmentControl<Post>(
+          stateKey: _currentPostKey,
+          defaultValue: Post.fakePosts().first,
+          builder: (_, Post post, EnvironmentState state) {
```

Comment:
Just so I understand, is this suggesting wrapping the arguments to `builder` in an object? There are a few things that give me pause about this approach:

1. This may just be a naming issue, but "model" generally tends to refer to a pure data object, while EnvironmentControlModel is not that.
2. I worry that this approach obscures a few things that might make the API sort of confusing. I'm not sure I'd expect setting `model.value` to have the side effects that it does. This also may simply be a naming issue.
3. `BuildContext` is generally passed directly, and accessing it via `model.context` might make it less clear that it's a normal context.

Reply:
My aim here isn't to wrap arguments, it's mostly to try to simplify the line `state.set(_currentPostKey, newPost);` because:
- The developer has already specified the `stateKey:` above, so why do they have to use it again here?
- The current `Post` value is dereferenced automatically and supplied as a parameter to the builder, so why does the  developer need to do this manually on the write path? The level of abstraction seems inconsistent here. Alternatively, you could not pass `currentValue` so that the dereferencing has to happen on read and write consistently.

BTW I'm not opposed to what you have right now - very happy to see you submit it and keep iterating. Alternatively you could pass three arguments - 
`builder: (BuildContext context, Post currentValue, Function(Post) updatePost)`.

Reply:
Resolving now that `EnvironmentControl` has been introduced.